ipatests/test_webui/task_range.py

```diff
@@ -67,50 +95,139 @@ def get_domain(self):
             domain = trusts[0]['cn']
         return domain
 
-    def get_data(self, pkey, size=50, add_data=None):
+    def get_data(self, pkey=None, form_data=None, **kwargs):
 
-        if not add_data:
-            add_data = self.get_add_data(pkey, size=size)
+        if not pkey:
+            pkey = 'itest-range-{}'.format(uuid.uuid4().hex[:8])
+
+        if form_data:
+            form_data.cn = pkey
+        else:
+            form_data = self.get_add_form_data(pkey, **kwargs)
 
         data = {
             'pkey': pkey,
-            'add': add_data,
+            'add': form_data.serialize(),
             'mod': [
-                ('textbox', 'ipaidrangesize', str(size + 1)),
+                ('textbox', 'ipaidrangesize', str(form_data.size + 1)),
             ],
         }
         return data
 
-    def get_add_data(self, pkey, range_type='ipa-local', size=50, shift=100, domain=None):
+    def get_add_form_data(self, pkey, range_type=LOCAL_ID_RANGE, size=50,
+                          domain=None, **kwargs):
+        """
+        Generate RangeAddFormData instance with initial data based on existing
+        ID ranges.
+        """
 
-        base_id = self.max_id + shift
+        shift = 100
+        base_id = kwargs.get('base_id', self.max_id + shift)
```

Comment:
Does the flow of `base_id` and `secondary_base_rid` need to be controlled by kwargs? Usage of kwargs makes it harder to analyze method's behavior from a signature. In this case simple `is not None` check with defaults will make the same job. Same for occurrences bellow.

Reply:
There are three reasons for that:
1) `base_rid` and `secondary_base_rid` are not so common arguments
2) kwargs allow to forward arguments to other methods easily
3) kwargs allow to increase number of options (field values in this case) without breaking signature.
But for sure I need to describe kwargs options for `get_add_form_data` in docstring